### PR TITLE
DBT-556 Added support for materializing Kudu table through impala adapter

### PR DESCRIPTION
**Pull request by niranjancdw**
## Describe your changes
 Added support for materialising Kudu table through impala adapter. 

## Internal Jira ticket number or external issue link
https://jira.cloudera.com/browse/DBT-556

## Testing procedure/screenshots(if appropriate):
https://gist.github.com/niranjancdw/cffbbd3dbe8126b9c96c25f645644c3d

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] I have formatted my added/modified code to follow pep-8 standards
- [X] I have checked suggestions from python linter to make sure code is of good quality.

### CODE REVIEW

**Comment by kennycontreras** [comment]
I think in that case this one could be skipped then? https://github.com/cloudera/dbt-impala/pull/200

**Comment by niranjancdw** [comment]
> I think in that case this one could be skipped then? #200

Yes, you can close that. We'll get this PR merged soon.

File: KUDU_INTEGRATION.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+### Requirements
+
+- In a CDP public cloud deployment, Kudu is available as one of the many Cloudera Runtime services within the Real-time Data Mart template.
+- To use Kudu, you can create a Data Hub cluster by selecting Real-time Data Mart template template in the Management Console.
+- Follow this [article](https://blog.cloudera.com/integrating-cloudera-data-warehouse-with-kudu-clusters) on integrating the created Kudu service with Impala CDW.  
+
+
+For general instructions, please follow [Readme](README.md) guidelines.
+
+## Supported features
```
Comment (niteshy) [inline KUDU_INTEGRATION.md:20]: can we add this table to Readme.md, as there is only one additional column.

Comment (niranjancdw) [reply KUDU_INTEGRATION.md]: Sure, added this change.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -61,6 +61,7 @@ When `dbt-impala` is installed this way, any changes you make to the `dbt-impala
 `dbt-impala` contains [functional](https://github.com/cloudera/dbt-impala/tree/master/tests/functional/) tests. Functional tests require an actual Impala warehouse to test against.
 
 - You can run functional tests "locally" by configuring a `test.env` file with appropriate `ENV` variables.
+- To run `Kudu functional tests` as part of the test suite when underlying storage is `Kudu`, please set the `ENV` variable `DISABLE_KUDU_TEST` to `false`.
```
Comment (niteshy) [inline CONTRIBUTING.md:64]: nit: Default it is set to True ie Kudu test are disabled.

File: README.md [hunk 3]
```diff
@@ -40,40 +41,40 @@ demo_project:
...
-|Authentication: LDAP|Yes| Yes |
-|Authentication: Kerberos|No| No |
+| Name | Base | Iceberg | Kudu |
+|------|------|---------|------|
+|Materialization: View|Yes| N/A | N/A |
+|Materialization: Table|Yes| Yes | Yes |
+|Materialization: Table with Partitions |Yes| Yes | No |
+|Materialization: Incremental - Append|Yes| Yes | Yes |
+|Materialization: Incremental - Append with Partitions |Yes| Yes | No |
+|Materialization: Incremental - Insert+Overwrite |Yes| Yes | Yes |
```
Comment (niteshy) [inline README.md:72]: Where are the test for `Insert+Overwrite` ? Most of the test are disabled.

Comment (niranjancdw) [reply README.md:72]: I was adding tests based on Impala, looks like there is no separate `Insert+Overwrite` test for Impala too, `Insert+Overwrite` is combined with `partition_key`. Let me know if a separate test is needed, I'll add for both.

File: CONTRIBUTING.md (hunk 2 above)
Comment (niranjancdw) [reply CONTRIBUTING.md]: Addressed.

File: README.md (hunk 3 above)
Comment (niteshy) [reply README.md:72]: please add the tests. If is going to take long time then please plan it as a part of separate PR. Just update the docs

**Review by niteshy — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (niranjancdw) [reply README.md]: Updated the docs now and created separate JIRAs to track the missing and disabled tests.

**Review by niteshy — APPROVED** [review]
Let's merge this. Create a follow-up ticket to handle the partition tables.

**Pull request merged**